pkg/api/handler/tx_finder.go

```diff
@@ -28,56 +28,71 @@ func (f txFinder) GetRawTxs(ctx context.Context, ids []string) ([]validator.RawT
 	var remainingIDs []string
 
 	// first get transactions from the handler
-	// improvement idea: implement getMany method
-	for _, id := range ids {
-		txBytes, _ := f.th.GetTransaction(ctx, id)
-		if txBytes != nil {
+	if source.Has(validator.SourceTransactionHandler) {
+		txs, _ := f.th.GetManyTransactions(ctx, ids)
+		for _, tx := range txs {
 			rt := validator.RawTx{
-				TxID:  id,
-				Bytes: txBytes,
-				// TODO: add block info?
+				TxID:    tx.TxID,
+				Bytes:   tx.Bytes,
+				IsMined: tx.BlockHeight > 0,
 			}
 
 			foundTxs = append(foundTxs, rt)
+		}
+
+		// add remaining ids
+		for _, id := range ids {
+			found := false
+			for _, tx := range foundTxs {
+				if tx.TxID == id {
+					found = true
+					break
+				}
+			}
 
-		} else {
-			remainingIDs = append(remainingIDs, id)
+			if !found {
+				remainingIDs = append(remainingIDs, id)
+			}
 		}
 	}
 
 	ids = remainingIDs[:]
 	remainingIDs = make([]string, 0)
 
 	// try to get remaining txs from the node
-	for _, id := range ids {
-		nTx, err := getTransactionFromNode(f.pc, id)
-		if err != nil {
-			// do we really need this info?
-			f.l.Warn("failed to get transaction from node", slog.String("id", id), slog.String("err", err.Error()))
+	if source.Has(validator.SourceNodes) {
+		for _, id := range ids {
+			nTx, err := getTransactionFromNode(f.pc, id)
```

Comment:
nit: Move this function `getTransactionFromNode` from `helpers.go` to `tx_finder.go`?

Reply:
Good point

Reply:
Moved